pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -415,6 +415,9 @@ func (pl *DynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
 						return nil, status
 					}
 				} else {
+					if !pl.enablePrioritizedList {
```

Comment:
Should we reject pods with `FirstAvailable` at PreEnqueue additionally when PrioritizedList is disabled? which would reduce unnecessary scheduling cycles consumed for those pods.

Reply:
That would be even better. Let me check that.

Reply:
Right, there was a reason (hadn't seen your second message when writing my initial reply).

Reply:
Yeah, it would be nice to be in PreEnqueue performance-wise though. Anyways, we can revisit all those PreEnqueue stuff later.

Reply:
This situation is going to be extremely rare (enable feature, create claim, disable feature) and temporary (will disappear as the feature matures, hopefully in a few releases).  It's good to catch it when it's just one if check in some existing code, but I am not sure whether it would be worth more complex changes in PreEnqueue.

I was about to write that when I saw that you had already retracted the suggestion. :smile: